### DISCUSSION TITLE
how do I remove the blank line before title

### DISCUSSION THREAD

**Discussion by duan-1** [zimfw/zimfw#499] [Q&A] [answered]
There has a blank line before title when I open Konsole. even I run 'clear' command, this line still here.
![image](https://user-images.githubusercontent.com/61895588/220254418-da653482-dbf3-4578-8adc-02ef32c6bd5b.png)

**Comment by antoineco** [comment, marked as answer]
This behavior comes from the Pure prompt, not Zim. Pure is a third-party prompt.

You can find a suggestion to achieve what you want at https://github.com/sindresorhus/pure/issues/509#issuecomment-641001784. However, this is hacky and unsupported, so I'd recommend forking the prompt and making it your own instead.

Remove the line below, then perform the following change to your config:

```patch
diff --git a/pure.zsh b/pure.zsh
index dab82aa..9521240 100644
--- a/pure.zsh
+++ b/pure.zsh
@@ -188,7 +188,7 @@ prompt_pure_preprompt_render() {
 
 	if [[ $1 == precmd ]]; then
 		# Initial newline, for spaciousness.
-		print
 	elif [[ $prompt_pure_last_prompt != $expanded_prompt ]]; then
 		# Redraw the prompt.
 		prompt_pure_reset_prompt
```

```diff
-zmodule sindresorhus/pure --source async.zsh --source pure.zsh
+zmodule duan-1/pure --source async.zsh --source pure.zsh
```

Apply with `zimfw uninstall && zimfw install`.

![Screenshot 2023-02-21 at 11 46 14](https://user-images.githubusercontent.com/3299086/220324130-fffaf4aa-4d55-4048-9ec8-788af90db001.png)

**Reply by duan-1** [reply]
But the default theme also has a blank line, this is the reason why I use pure. Then I found pure also has a blank line, it made me think is the zim's promble.
![image](https://user-images.githubusercontent.com/61895588/220389771-7ad0341f-df30-4e8f-b677-bb305ff2053e.png)

**Reply by antoineco** [reply]
That's a good point. `asciiship` draws some inspirations from `pure` and `spaceship`, and just like them its prompt starts with an empty line: [asciiship.zsh-theme#L50-L52](https://github.com/zimfw/asciiship/blob/2e349ec842e9e5417d5ecebd37509673c97ef28a/asciiship.zsh-theme#L50-L52).

Same procedure if you want to get rid of it: fork, remove the leading newline character at line 50, and use the forked version in your config.

**Reply by duan-1** [reply]
thx a lot. I used powerlevel10k finally, it can also achieve the same effect as pure. No more blank line now!  :D
